src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -559,12 +613,15 @@ impl<'a, 'tcx> LinkCollector<'a, 'tcx> {
 
         match root_res {
             Res::Primitive(prim) => self.resolve_primitive_associated_item(prim, ns, item_name),
+            Res::Def(DefKind::TyAlias, did) => {
+                // Resolve the link on the type the alias points to.
+                // FIXME: if the associated item is defined directly on the type alias,
+                // it will show up on its documentation page, we should link there instead.
```

Comment:
Could we do that by checking `inherent_impls(did)` before calling `resolve_associated_item`?

Reply:
If I'm right in https://github.com/rust-lang/rust/pull/86334#discussion_r653912376, `inherent_impls` won't return anything for a type alias. I think `rustdoc` is able to display the implementations for type aliases because it does its own visit of the HIR to collect them. Maybe there's a way to use that here, but when I tried the cache was still empty, I'm assuming it gets populated later in the process.

Reply:
Got it. As long as the link works I'm not super concerned about whether it goes to the type alias or not.